### PR TITLE
mgr/dashboard: fix frontend e2e tests

### PR DESCRIPTION
**Pull request by tspmelo**
Updated protractor configuration: now it's possible to run the tests inside a
docker container, added params for the username and password and added login
during the prepare phase.

Chrome is required to run the e2e tests. This can be achieved by installing
Chrome in your system or by using a docker container with Chrome installed.

Changed some timer calls, so they are runned outside of angular. This was
preventing protractor from running the tests.

`Signed-off-by: Tiago Melo <tmelo@suse.com>`

### CODE REVIEW

File: src/pybind/mgr/dashboard/HACKING.rst [hunk 1]
```diff
@@ -72,8 +72,21 @@ Run ``npm run test`` to execute the unit tests via `Karma
...
+Note:
+  You need Chrome installed and to have the dashboard frontend built.
+
+Docker:
+  Alternatively, if you don't have/want Chrome installed in your system,
+  you can use a docker image to run the e2e tests.
+
+  .. code:: bash
+
+    # docker run -d -v $(pwd):/workdir --net=host --name angular-e2e-container rogargon/angular-e2e
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/HACKING.rst]: Do these commands have to be run as the root user? If not, consider replacing the command prompt with `$` instead.

Comment (tspmelo) [reply src/pybind/mgr/dashboard/HACKING.rst]: docker commands require root.

**Comment by tspmelo** [comment]
Updated `HACKING.rst` e2e section and improved `run-frontend-e2e-tests.sh` script.

File: src/pybind/mgr/dashboard/run-frontend-e2e-tests.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  esac
+done
+
+DASH_DIR=`pwd`
+
+cd ../../../../build
+BUILD_DIR=`pwd`
+
+if [ "$BASE_URL" == "" ]; then
+    ../src/stop.sh
```
Comment (rjfd) [inline src/pybind/mgr/dashboard/run-frontend-e2e-tests.sh]: when no vstart cluster is running, this command/script will block indefinitely. You should only stop the cluster in the end of the script, even if there was some error during the tests execution.

Comment (tspmelo) [reply src/pybind/mgr/dashboard/run-frontend-e2e-tests.sh]: Done

File: src/pybind/mgr/dashboard/frontend/e2e/cluster/monitors.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { browser } from 'protractor';
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/frontend/e2e/cluster/monitors.e2e-spec.ts]: Unused import.

File: src/pybind/mgr/dashboard/frontend/e2e/cluster/osds.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { browser } from 'protractor';
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/frontend/e2e/cluster/osds.e2e-spec.ts]: Unused import.

File: src/pybind/mgr/dashboard/frontend/e2e/helper.po.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { $, browser, by, element } from 'protractor';
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/frontend/e2e/helper.po.ts]: `by`and `element` are not used.

File: src/pybind/mgr/dashboard/frontend/e2e/cluster/osds.po.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { $, browser, by, element } from 'protractor';
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/frontend/e2e/cluster/osds.po.ts]: `by`and `element` are not used.

**Comment by tspmelo** [comment]
Removed unused imports.

**Comment by Devp00l** [comment]
Skript doesn't work for me:

`# ./run-frontend-e2e-tests.sh -d chrome -r "http://localhost:4200/#"`
Line 53: /home/albatros/projects/openattic/cephmgr/ceph-dev/build/src/pybind/mgr/dashboard/node-env/bin/activate: File or Directory not found

`# ./run-frontend-e2e-tests.sh -d chrome`
/.../ceph-dev/build/bin/ceph-conf: error while loading shared libraries: libibverbs.so.1: cannot open shared object file: No such file or directory
dirname: missing operand

**Comment by rjfd** [comment]
@Devp00l did you built ceph before running the script? the first error you mention should be fixed by building ceph. The second error is solved by installing the missing library in your system.

**Comment by rjfd** [comment]
@tspmelo I think there are still two things that we need to improve in this script:
* when an error occurs during the execution of the script the vstart cluster should be stopped (not only when it finishes successful)
* when running with docker the "frontend/dist" directory becomes owned by the root user. Can you build the frontend in a temporary directory just for the use of the docker container?

**Comment by tspmelo** [comment]
@rjfd now the script will call stop if docker or chrome fails. 
I also changed the call to `npm install` and `build` so it is run outside docker.

**Comment by tspmelo** [comment]
jenkins retest this please

**Pull request merged**